src/compiler/parse_table.cc

```diff
@@ -154,7 +154,8 @@ void ParseState::each_referenced_state(function<void(ParseStateId *)> fn) {
       if (action.type == ParseActionTypeShift || action.type == ParseActionTypeRecover)
         fn(&action.state_index);
   for (auto &entry : nonterminal_entries)
-    fn(&entry.second);
+    if (entry.second != (ParseStateId)(-1))
```

Comment:
I don't think we need this conditional; the only things we write into this map are valid state numbers.